### DISCUSSION TITLE
create_multipart_upload does not publish content-length

### DISCUSSION THREAD

**Discussion by allada** [awslabs/aws-sdk-rust#1011] [Q&A]
### Describe the bug

Some S3 API providers require the `content-length` header to be published when using the S3 API and `create_multipart_upload()`. Specifically GCP is an example of this, therefore it is not possible to use `aws-sdk-rust` with GCP.

### Expected Behavior

`create_multipart_upload()` should send a `content-length` header.

### Current Behavior

`content-length` header is not published.

### Reproduction Steps

Try to use the `create_multipart_upload()` with GCP's API.

### Additional Information/Context

Possible more context: https://github.com/TraceMachina/nativelink/issues/491

### Version

```text
aws-sdk-s3 = "0.35.0"
```

### Environment details (OS name and version, etc.)

N/A

**Comment by jdisanti** [comment]
Do they require content-length on CreateMultipartUpload, or on UploadPart? The UploadPart operation has a `content_length` field you can set if it's the latter.

If it's needed for CreateMultipartUpload, you can call `.customize().mutate_request()`, which will allow you to add a header to the HTTP request before it is sent. Content-length isn't modeled for CreateMultipartUpload, so it's not something we would ever add there unless S3 modeled it.
